azure/LongRunningOperations.md

```diff
@@ -0,0 +1,69 @@
+# Long-Running Operations
+
+Long-running operations are an API design pattern that should be used when the processing of
+an operation may take a significant amount of time -- longer than a client will want to block
+waiting for the result.
+Azure allows for two forms of this design pattern: resource-based long-running operations (RELO),
+which is the preferred pattern, and stepwise long-running operations.
+
+In both patterns, the processing of the operation is initiated by one API call and the client
+obtains the results of the operation from a subsequent API call.
+Here we illustrate the sequence of API calls involved in each of these patterns.
+
+## Resource-based long-running operations
+
+In the RELO pattern, the resource that is the target of the operation contains a `status` field
+that holds the status of an outstanding or last completed operation.
+This means that the client can use a standard "get" operation on the resource to determine the
+status of an operation it initiated.  The flow looks like this:
+
+![The RELO flow](./relo.jpg)
+
+1. The client sends the initial request to the resource to initiate the long-running operation.
+This initial request could be a PUT, PATCH, POST, or DELETE method.
+
+2. The resource validates the request and initiates the operation processing.
+It sends a response to client with a `200-OK` HTTP status code (or `201-Created` if the operation
+is a create operation) and a representation of the resource where the `status` field is set
+to a value indicating that the operation processing has been started.
+
+3. The client then issues a GET request to the resource to determine if the operation processing
+has completed.
+
+4. The resource responds with a representation of the resource. While the operation is still being
+processed, the status field will contain a "non-terminal" value, like "processing".
+
+5. After the operation processing has completed, a GET request from the client will receive a response
+where the status field contains a "terminal" value -- `succeeded`, `failed`, or `cancelled` --
```

Comment:
Inconsistent casing has been used "Failed" vs "failed".
Also, i think it's "Success"; not "succeeded".
Also, I think cancelled with 1 L and 2 Ls has also been used.
All this needs to be cleaned up.

Reply:
I did some research on "Success" vs "Succeeded":
- The [ARM guidelines](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#provisioningstate-property) say "Succeeded"
- Looking just at resource-manager API definitions in azure-rest-api-specs:
  - "Succeeded" appears as an enum value (well -- on a line by itself) 1339 times in 694 files
  - "Success" appears as an enum value 78 times in 68 files

I think it is also desirable for enum values to be consistent in their part of speech -- e.g. all verbs or all nouns.  "Success" is a noun.  I believe that "Succeeded", "Failed", and "Canceled" are all intransitive verbs.

So I think we want to use "Succeeded".

I will make the other corrections.